pedl_train.py

```diff
@@ -0,0 +1,423 @@
+    ) -> Dict[str, Any]:
+
+        batch_size = 16
+
+        results, maps = test.test(
+            cfg,
+            data,
+            batch_size=batch_size * 2,
+            img_size=img_size_test,
+            model=model,
+            conf_thres=0.1,  # 0.1 for speed
+            # 1e-3 if opt.evolve or (final_epoch and is_coco) else 0.1,
+            iou_thres=0.6,
+            single_cls=opt.single_cls,
+            dataloader=data_loader,
+        )
+        keys = [
+            "P",
+            "R",
+            "mAP_at_0.5",
+            "F1",
+            "Some_number_1",
+            "Some_number_2",
+            "Some_number_3",
+        ]
+
+        return dict(zip(keys[:4], results))
+
+
+if __name__ == "__main__":
+
+    # Hard code default arguments
+    opt = argparse.Namespace(
+        accumulate=4,
+        adam=True,
+        arc="default",
+        batch_size=16,
+        bucket="",
+        cache_images=False,
+        cfg="cfg/yolov3-spp.cfg",
+        data="data/coco2017.data",
+        device="",
+        epochs=273,
+        images=117263,
+        evolve=False,
+        img_size=[416],
+        multi_scale=False,
+        name="",
+        nosave=True,  # default False
+        notest=True,  # default False
+        rect=False,
+        resume=False,
+        single_cls=False,
+        var=None,
+        weights="weights/ultralytics68.pt",
+    )
+
+    hyp = {
+        "giou": 3.54,  # giou loss gain
+        "cls": 37.4,  # cls loss gain
+        "cls_pw": 1.0,  # cls BCELoss positive_weight
+        "obj": 49.5,  # obj loss gain (*=img_size/320 if img_size != 320)
+        "obj_pw": 1.0,  # obj BCELoss positive_weight
+        "iou_t": 0.225,  # iou training threshold
+        "lr0": 0.000579,  # initial learning rate (SGD=5E-3, Adam=5E-4) #0.00579
+        "lrf": -4.0,  # final LambdaLR learning rate = lr0 * (10 ** lrf)
+        "momentum": 0.937,  # SGD momentum
+        "weight_decay": 0.000484,  # optimizer weight decay
+        "fl_gamma": 0.5,  # focal loss gamma
+        "hsv_h": 0.0138,  # image HSV-Hue augmentation (fraction)
+        "hsv_s": 0.678,  # image HSV-Saturation augmentation (fraction)
+        "hsv_v": 0.36,  # image HSV-Value augmentation (fraction)
+        "degrees": 1.98,  # image rotation (+/- deg)
+        "translate": 0.05,  # image translation (+/- fraction)
+        "scale": 0.05,  # image scale (+/- gain)
+        "shear": 0.641,
+    }  # image shear (+/- deg)
+
+    cfg = opt.cfg
+    data = opt.data
+    img_size, img_size_test = (
+        opt.img_size if len(opt.img_size) == 2 else opt.img_size * 2
+    )  # train, test sizes
+    images = opt.images
+    epochs = opt.epochs  # 500200 batches at bs 64, 117263 images = 273 epochs
+    batch_size = opt.batch_size
+    accumulate = opt.accumulate  # effective bs = batch_size * accumulate = 16 * 4 = 64
+    weights = opt.weights  # initial training weights
```

Comment:
is there a good reason to create a bunch of redundant variables? I would default to removing all of these variables and using `opt` directly throughout if not.

Reply:
Yeah - again this piece is copy-pasted from the orginal repo - I don't know why they did that. It is a bit weird I agree https://github.com/ultralytics/yolov3/blob/master/train.py#L56

Reply:
cleaned it up.